include/iceoryx_dds/dds/fast_data_reader.hpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2020 - 2021 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_DDS_DDS_FAST_DATA_READER_HPP
+#define IOX_DDS_DDS_FAST_DATA_READER_HPP
+
+#include "iceoryx_dds/dds/data_reader.hpp"
+
+#include <atomic>
+#include <fastdds/dds/subscriber/DataReader.hpp>
+#include <fastdds/dds/topic/Topic.hpp>
+
+namespace iox
+{
+namespace dds
+{
+/// @brief Implementation of the DataReader abstraction using the fastdds implementation.
+class FastDataReader : public DataReader
+{
+  public:
+    FastDataReader() = delete;
+
+    /// @brief Constructor to set fast data reader object from given IDs
+    /// @param[in] serviceId ID of the service
+    /// @param[in] instanceId ID of the instance of the service
+    /// @param[in] eventId ID of the event
+    FastDataReader(const capro::IdString_t serviceId,
```

Comment:
It would be much more efficient to use a reference here. Please rewrite it into `const capro::IdString_t & serviceId` for all parameters.